tests/population/test_rasterpop.py

```diff
@@ -0,0 +1,510 @@
+    longer the precise extremeties. This means the grids are no longer with the
+    bounds of the re-converted urban centre geometry. Therefore, only for this
+    unit test, a buffer is applied to the urban centre before conversion (just
+    1m). This is enough to negate the floating point/rounding errors and bring
+    the grids back within the converted urban centre boundary. This step is
+    not needed for the area of interest since floating point/round errors can
+    not lead to a conflicting scenario since there is always sufficent overlap
+    into the cells.
+
+    """
+    # convert aoi to EPSG:4326
+    aoi_4326_gdf = gpd.GeoDataFrame(
+        geometry=[xarr_1_aoi[0]], crs="ESRI:54009"
+    ).to_crs("EPSG:4326")
+
+    # convert uc to EPSG:4326 - see note for more details.
+    uc_4326_gdf = gpd.GeoDataFrame(geometry=[xarr_1_uc[0]], crs="ESRI:54009")
+    uc_4326_gdf["geometry"] = uc_4326_gdf.geometry.buffer(1, join_style=2)
+    uc_4326_gdf = uc_4326_gdf.to_crs("EPSG:4326")
+
+    return aoi_4326_gdf.loc[0, "geometry"], uc_4326_gdf.loc[0, "geometry"]
+
+
+class TestRasterPop:
+    """A class to test population.RasterPop methods."""
+
+    def test__raster_pop_internal_methods(
+        self,
+        xarr_1_fpath: str,
+        xarr_1_aoi: tuple,
+        xarr_1_uc: tuple,
+    ) -> None:
+        """Test all the internal methods of the RasterPop call.
+
+        Test the performance of the base internal methods. This test is written
+        in this way since the internal methods are dependent on one another,
+        and this permits unit testing of each internal stage.
+
+        Parameters
+        ----------
+        xarr_1_fpath : str
+            Filepath to dummy data GeoTIFF file. Output from `xarr_1_fpath`
+            fixture.
+        xarr_1_aoi : tuple
+            A tuple containing the area of interest polygon for xarr_1 and a
+            dictionary of expected method outputs after applying the area of
+            interest polygon. For more information on the valid keys of this
+            dictionary see the docstring of the `xarr_1_aoi` fixture.
+        xarr_1_uc : tuple
+            A tuple containing the urban centre polygon for xarr_1 and a
+            dictionary of expected method outputs after applying the urban
+            centre polygon. For more information on the valid keys of this
+            dictionary see the docstring of the `xarr_1_uc` fixture.
+
+        """
+        # print fpath where input data resides in tmp folder
+        # useful when using -rP flag in pytest to see directory
+        print(f"Temp file path for tif input: {xarr_1_fpath}")
+
+        rp = RasterPop(xarr_1_fpath)
+
+        # call and test read and clip method asserting post clip expectation
+        rp._read_and_clip(aoi_bounds=xarr_1_aoi[0])
+        assert np.array_equal(
+            rp._xds.to_numpy(), xarr_1_aoi[1]["post_clip"], equal_nan=True
+        )
```

Comment:
Assertion failure messages in test suite. Helpful in diagnosing the source of breaking changes. Agreed to raise under a technical debt label.